include/zephyr/sys/fdtable.h

```diff
@@ -148,6 +148,15 @@ void *z_get_fd_obj(int fd, const struct fd_op_vtable *vtable, int err);
 void *z_get_fd_obj_and_vtable(int fd, const struct fd_op_vtable **vtable,
 			      struct k_mutex **lock);
 
+/**
+ * @brief Get lock from file descriptor.
+ *
+ * @param fd File descriptor previously returned by z_reserve_fd()
+ *
+ * @return Mutex lock or NULL
+ */
+struct k_mutex *z_get_fd_lock(int fd);
```

Comment:
It might be possible to avoid adding this.

Reply:
It is possible if we modify `z_get_fd_obj_and_vtable()` to conditionally return `vtable`

Reply:
Not sure at the moment.

It might be better to simply do the same thing that would be done with a `FILE *`, which is call `zvfs_flockfile()`, `zvfs_ftrylockfile()`, and `zvfs_funlockfile()`.

With an `int` file descriptor, could do something like `zvfs_lockfile()`, `zvfs_trylockfile()`, and `zvfs_unlockfile()`.